### ISSUE TITLE
[ATable] RowAction Component

### ISSUE THREAD

**Issue by agritheory** [agritheory/stonecrop#52]
![image](https://user-images.githubusercontent.com/18033666/210903628-2331392b-283a-4b54-a69c-b34cc53e7704.png)

- [ ] Expand/ collapse - use same multiplication sign as Fieldset with same animation
- [ ] Delete ␡ [link](https://www.utf8icons.com/character/9249/symbol-for-delete)
- [ ] Insert Above ↥ [link](https://www.utf8icons.com/character/8613/upwards-arrow-from-bar)
- [ ] Insert Below ↧ [link](https://www.utf8icons.com/character/8615/downwards-arrow-from-bar)
- [ ] Duplicate ⫻ [link](https://www.utf8icons.com/character/11003/triple-solidus-binary-relation)
- [ ] [Move / click and drag](https://github.com/agritheory/stonecrop/issues/129) ↹ [link](https://www.utf8icons.com/character/8633/leftwards-arrow-to-bar-over-rightwards-arrow-to-bar)

Open to suggestions on the icons and all of them should have an alt text on hover
We probably also want key bindings

**Comment by agritheory** [comment]
Draggable VueUse helper: https://vueuse.org/core/useDraggable/

**Comment by agritheory** [comment]
@Alchez Probably time to revist this component
@crabinak Let's discuss icons and the UX of this at our next meeting, we could use your expertise here

**Comment by crabinak** [comment]
@agritheory Here are a few sets of icons I put together. Let me know if any of these stand out to you. 

01:
![ui-icons-01](https://github.com/agritheory/stonecrop/assets/166232142/4f94b9f3-96f1-4cd9-a1ce-88f3806be88e)
02:
![ui-icons-02](https://github.com/agritheory/stonecrop/assets/166232142/2268bfd4-fa0b-4d45-8102-41db1038de60)
03:
![ui-icons-03](https://github.com/agritheory/stonecrop/assets/166232142/3ed47708-15a6-41ce-a5da-b4554069c703)
04:
![ui-icons-04](https://github.com/agritheory/stonecrop/assets/166232142/12c2ff6c-f361-4c99-bb4b-1e5f6e90897a)

**Comment by crabinak** [comment]
@agritheory Here are some updated icons. I have two versions here: one with a thicker outline and rounded ends, and the other is thinner with square ends. I included two options of the Delete icon which are basically inverted versions of each other. 
![ui-icons-05](https://github.com/agritheory/stonecrop/assets/166232142/bbedeb84-a6f5-4b9e-9852-00abbc66799e)
![ui-icons-06](https://github.com/agritheory/stonecrop/assets/166232142/ccc1fa29-bc15-45ab-8936-b2e384acffc4)

**Comment by agritheory** [comment]
@crabinak these look great. I think I favor the thin ones. Let's get some consensus. @Alchez @fproldan @bhattdevarsh what do you guys think?

**Comment by Alchez** [comment]
@agritheory @crabinak I prefer the thin ones as well <sub>(and also the round versions of the insert above and below icons)</sub>
